tests/models/test_amp.py

```diff
@@ -199,7 +199,9 @@ def configure_optimizers(self):
     assert str(trainer.amp_backend) == "AMPType.APEX"
     trainer.fit(model)
     assert trainer.state.finished, f"Training failed with {trainer.state}"
-    assert bwd_mock.call_count == 10
+    # `max_steps` is fulfilled in the third batch first optimizer, but we don't check the loop
+    # `done` condition until all optimizers have run, so the number of backwards is higher than `max_steps`
+    assert bwd_mock.call_count == 6
```

Comment:
`assert _batches_that_stepped == 10` would hold?

Reply:
No, `assert _batches_that_stepped == 3`.

As described in the comment, `max_steps=5` is fulfilled in the third batch, first optimizer.

So we do 1 extra backward because the loop done is not checked yet.
(does not impact batches_that_stepped)

This is a breaking change as described in the PR intro:

> The Trainer arguments {min,max}_steps compare with the new global_step value so they suffer from the same breaking changes. In the case of multiple optimizers or TBPTT users will need to adjust them.